paasta_itests/steps/setup_chronos_job_steps.py

```diff
@@ -30,12 +30,21 @@ def job_exists(context, service, instance):
     assert len(matching_jobs) == 1
 
 
-@when(u'we run setup_chronos_job for service_instance "{service_instance}"')
-def run_setup_chronos_job(context, service_instance):
+@when(u'we run setup_chronos_job for service_instance "{service_instance}", expecting exit code {expected_exit_code}')
+def run_setup_chronos_job(context, service_instance, expected_exit_code):
     cmd = "../paasta_tools/setup_chronos_job.py %s -d %s" % (service_instance, context.soa_dir)
     exit_code, output = _run(cmd)
     context.exit_code, context.output = exit_code, output
 
+    try:
+        assert context.exit_code == int(expected_exit_code), \
```

Comment:
I'm not convinced we should be doing assertions in a '@when' - they should be in a '@then' 

In other places we have the execution of a script as a '@then', because we don't need to do anything after. See [paasta metastatus](https://github.com/Yelp/paasta/blob/3a7a5043b537c2f24c954d9d4b2dd2e061a97fb7/paasta_itests/steps/paasta_metastatus_steps.py#L63-L80) for example. But this isn't the case for this - we often do setup_chronos_job then check for some other thing in a '@then'.